specs/unmanage_cluster.adoc

```diff
@@ -0,0 +1,373 @@
+collectd and tendrl-gluster-integration
+
+* Mark the cluster flag `is_managed` as `False` so that the cluster could be
+listed as un-managed in UI dashboards and all the possible actions could be
+disabled for it
+
+* Delete cluster entity details from tendrl central store
+
+* Archive the graphite (monitoring) data for the cluster in archive location so
+the grafana dashboards dont list the cluster and its entities anymore
+
+* Delete the grafana alert dashboards for the cluster and its dependent entities
+
+The logic here goes like
+
+** Start a flow in node-agent on tendrl server node for un-manage cluster
+
+** The first atom of the above flow invokes child jobs on the storage node's
+node-agent to stop tendrl specific services and marking them disabled
+
+** In the main atom of the un-manage cluster flow remove if any etcd details for
+the cluster and then mark the cluster is_managed flag as `False`
+
+** One of the atoms now un-manage cluster flow, invokes a flow in
+monitoring-integration to archive the graphite data for the cluster
+
+** Finally another atom invokes a flow in monitoring-integration to remove the
+grafana alert dashboards for the cluster and its dependent entities
+
+So the structure of the un-manage cluster flow would look something as below
+
+```
+UnmanageCluster:
+  tags:
+    - "tendrl/monitor"
+  atoms:
+    - tendrl.objects.Cluster.atoms.StopMonitoringServices
+    - tendrl.objects.Cluster.atoms.StopIntegrationServices
+    - tendrl.objects.Cluster.atoms.DeleteClusterDetails
+    - tendrl.objects.Cluster.atoms.DeleteMonitoringDetails
+  help: "Unmanage a Gluster Cluster"
+  enabled: true
+  inputs:
+    mandatory:
+      - TendrlContext.integration_id
+  run: tendrl.flows.UnmanageCluster
+  type: Update
+  uuid: 2f94a48a-05d7-408c-b400-e27827f4efed
+  version: 1
+```
+
+* While import flow in progress the values of `current_job` and `status`
+should be set with `{'job_id': 'import job id', 'job_name': 'ImportCluster',
+'status': 'in_progress'}` id and `Importing` respectively
+
+* Once import flow is successful the value of `status` would be set as `done`
+
+* If import flow fails the value of `status` would be set as `failed`
+
+* While un-manage flow in progress the values of `current_job` and `status`
+should be set with `{'job_id': 'unmanage job id', 'job_name': 'ImportCluster',
+'status': 'in_progress'}` and `Unmanaging` respectively
+
+* Once un-manage flow is successful the value of `status` would be set as `done`
+
+* If un-manage flow fails the value of `status` would be set as `failed`
+
+* If an import cluster fails tendrl UI needs to keep import cluster option open
+and if user selects the option, it should throw a dialog telling about the
+previous import failure and if user confirms to go ahead about un-manage and
+then import the cluster, UI should submit an un-manage cluster first. If the
+un-manage cluster task succeeds, then UI should submit a import for the same
+cluster
```

Comment:
I have a question here. If user chooses to click on import button again how will the `View Details` link work? Will it link first to `Task Details` of un-manage task and later details of import task?

Reply:
@ltrilety 
If a user chooses to click on Import button, the View Details will show the link to the last task run.  

If an unmanage ran successfully, then import ran successfully/unsuccessfully, it will show the import.

If an unmanage did not run successfully (then import does not run), it will show the unmanage.